lib/rubocop/cop/capybara/rspec/predicate_matcher.rb

```diff
@@ -0,0 +1,269 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Capybara
+      module RSpec
+        # A helper for `inflected` style
+        module InflectedHelper
+          extend NodePattern::Macros
+
+          EXPLICIT_MATCHER = %w[css selector style xpath].each.map do |suffix|
+            "matches_#{suffix}?"
+          end.freeze
+          MSG_INFLECTED = 'Prefer using `%<matcher_name>s` matcher over ' \
+                          '`%<predicate_name>s`.'
+
+          private
+
+          def check_inflected(node)
+            predicate_in_actual?(node) do |predicate, to, matcher|
+              msg = message_inflected(predicate)
+              add_offense(node, message: msg) do |corrector|
+                remove_predicate(corrector, predicate)
+                corrector.replace(node.loc.selector,
+                                  true?(to, matcher) ? 'to' : 'not_to')
+                rewrite_matcher(corrector, predicate, matcher)
+              end
+            end
+          end
+
+          # @!method predicate_in_actual?(node)
+          def_node_matcher :predicate_in_actual?, <<~PATTERN
+            (send
+              (send nil? :expect
+                {
+                  (block $(send !nil? #predicate? ...) ...)
+                  $(send !nil? #predicate? ...)
+                }
+              )
+              ${:to :to_not :not_to}
+              $#boolean_matcher?
+            )
+          PATTERN
+
+          # @!method be_bool?(node)
+          def_node_matcher :be_bool?, <<~PATTERN
+            (send nil? {:be :eq :eql :equal} {true false})
+          PATTERN
+
+          # @!method be_boolthy?(node)
+          def_node_matcher :be_boolthy?, <<~PATTERN
+            (send nil? {:be_truthy :be_falsey :be_falsy :a_truthy_value :a_falsey_value :a_falsy_value})
+          PATTERN
+
+          def boolean_matcher?(node)
+            if cop_config['Strict']
+              be_boolthy?(node)
+            else
+              be_bool?(node) || be_boolthy?(node)
+            end
+          end
+
+          def predicate?(sym)
+            EXPLICIT_MATCHER.include?(sym.to_s)
```

Comment:
Maybe pre-convert those to `to_sym` and skip `to_s` here?

Reply:
Indeed it is, I updated this PR 👍🏻